packages/admin/admin-generator/src/commands/generate/generateGrid/generateGrid.ts

```diff
@@ -209,14 +209,36 @@ export function generateGrid(
     const gridQueryType = findQueryTypeOrThrow(gridQuery, gqlIntrospection);
 
     const createMutationType = findMutationType(`create${gqlType}`, gqlIntrospection);
+    const updateMutationType = findMutationType(`update${gqlType}`, gqlIntrospection);
 
     const hasDeleteMutation = !!findMutationType(`delete${gqlType}`, gqlIntrospection);
     const hasCreateMutation = !!createMutationType;
+    const hasUpdateMutation = !!updateMutationType;
 
     const allowCopyPaste = (typeof config.copyPaste === "undefined" || config.copyPaste === true) && !config.readOnly && hasCreateMutation;
     const allowAdding = (typeof config.add === "undefined" || config.add === true) && !config.readOnly;
     const allowEditing = (typeof config.edit === "undefined" || config.edit === true) && !config.readOnly;
     const allowDeleting = (typeof config.delete === "undefined" || config.delete === true) && !config.readOnly && hasDeleteMutation;
+    const allowRowReordering = typeof config.rowReordering?.enabled !== "undefined" && config.rowReordering?.enabled && hasUpdateMutation;
+
+    const updateInputArg = updateMutationType?.args.find((arg) => arg.name === "input");
+    if (allowRowReordering && updateInputArg) {
+        const inputType = findInputObjectType(updateInputArg, gqlIntrospection);
+        if (!inputType) throw new Error("Can't find update input type");
+        if (!inputType.inputFields?.find((field) => field.name === "position")) {
+            throw new Error("Position field is needed when using 'rowReordering'");
+        }
+    }
+
+    const hasRowReorderingOnDragField =
+        allowRowReordering && typeof config.rowReordering?.dragPreviewField !== "undefined" && !!config.rowReordering?.dragPreviewField;
```

Comment:
Aren't these two checks basically the same?

```suggestion
        allowRowReordering && typeof config.rowReordering?.dragPreviewField !== "undefined";
```

Reply:
[d66824b](https://github.com/vivid-planet/comet/pull/3519/commits/d66824bc664238eefb24a8d66c4a69e12838b892)

yes you are right